### DISCUSSION TITLE
Code Action: Inline Variable

### DISCUSSION THREAD

**Discussion by elenakrittik** [microsoft/pylance-release#5704] [Enhancement]
Similar to rust-analyzer's [inline-local-variable](https://rust-analyzer.github.io/manual.html#inline_local_variable) and [inline_type_alias](https://rust-analyzer.github.io/manual.html#inline_type_alias).

For example, if i put my cursor at the expression in f-string here:

```py
name = "John"

print(f"Hello, {name}!")
```
I want to be able to inline `name` into the f-string, so that the code turns into this:

```py
print(f"Hello, John!")
```

But if `name` was used by something else as well, i want to keep it (to not break the code, of course):

```py
name = "John"

print(f"Hello, John!")

# ...

print(f"Bye, {name}!")
```

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
